x/bank/keeper/msg_server.go

```diff
@@ -15,6 +14,17 @@ type msgServer struct {
 	Keeper
 }
 
+func (k msgServer) UpdateDenomMetadata(goCtx context.Context, msg *types.MsgUpdateDenomMetadata) (*types.MsgUpdateDenomMetadataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.Keeper.SetDenomMetaData(ctx, *msg.Metadata)
```

Comment:
There should be a check before this to make sure that the `msg.FromAddress == k.authority`. Basically, this message can only be sent by the gov module.